### ISSUE TITLE
Incorrect entity identified

### ISSUE THREAD

**Issue by kl-wong** [wit-ai/wit#750]
I got an entity prefix with a list of predefined acronym and the lookup strategies set to keywords.
Here is some acronym keywords: ALL, MAN, GE, STT, PWQ

Question: **Ge**t **all** **STT** record
Current behaviour:
prefix: GE
prefix: ALL
prefix: STT

Expected behaviour:
prefix: STT

Question: How **man**y **PWQ** record added
Current behaviour:
prefix: MAN
prefix: PWQ

Expected behaviour:
prefix: PWQ

How can I train the app to have the expected behaviour?

**Comment by patapizza** [comment]
Hi @kl-wong,

Please use the provided template.
What is your app id?

**Comment by kl-wong** [comment]
**Do you want to request a feature, report a bug, or ask a question about wit?**
Question:
I got an entity prefix with a list of predefined acronym and the lookup strategies set to keywords.
Here is some acronym keywords: ALL, MAN, GE, STT, PWQ

When asking question like
Example1: "Get all STT record"
Example2: "How many PWQ record added" 

For part of those words "Get", "all", "many", the app always return as an entity.
How can I train the app to eliminate those confusion?

In addition I got another issue "**Ongoing training status #747**" set to closed but the issue haven't resolved. Should I create another issue?
Basically the problem is about entity insights. Most of the entity I can see the insights but the entity prefix still haven't shown.

**What is the current behavior?**
Example1 : 
text: **Ge**t **all** STT record
prefix: GE
prefix: ALL
prefix: STT

Example2: 
text: How **man**y PWQ record added
prefix: MAN
prefix: PWQ

**What is the expected behavior?**
Example1: prefix: STT
Example2: prefix: PWQ

**What is the App ID where you are experiencing this issue (if applicable)?**
LNSRV2DYIHMUJCU2SNXLCUMTK6L3ROXK

**Comment by patapizza** [comment]
@kl-wong This is the app token. Please share the app id (and reset your app token).

**Comment by kl-wong** [comment]
Oh....! I am so sorry. Wrong field copied.
 App ID: 59a498fb-c649-471f-89b4-2b64a0658dff

**Comment by patapizza** [comment]
It looks like you haven't set a value `STT` nor `PWQ` for the entity `prefix`.
Try to add those, and feel free to reopen if that doesn't help.

**Comment by kl-wong** [comment]
@patapizza I have used other prefix in my example. As you can see "How many" and "Get all" are just expression but the app still pick up those keyword as entity.
![screen shot 2017-09-01 at 9 59 41 am](https://user-images.githubusercontent.com/31428678/29950024-63d1c4a0-8efc-11e7-9b6b-42186c3efba4.png)

**Comment by patapizza** [comment]
@kl-wong Oh I see. Sorry for the confusion. I'm not sure how we can handle that, and be somewhat flexible at the same time (e.g. catching plural forms). Would a regex on your side work instead of the `prefix` entity?

**Comment by kl-wong** [comment]
@patapizza I think I probably add Synonyms for the prefix and leave the keywords as it is. That should solve the problem.
